File: first_publisher_subscriber/src/yao_node.cpp
/*
roscore
rosrun first_publisher_subscriber yao_node
rostopic list
rostopic hz kaihei_room
rostopic echo kaihei_room
*/

#include <ros/ros.h>
#include <std_msgs/String.h>
ros::Publisher publisher;

void yao_callback(std_msgs::String msg)
{
    if(msg.data == "Play a game?")
    {
        std_msgs::String response_msg;
        response_msg.data = "+++++++++";
        publisher.publish(response_msg);
    }
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "yao_node");

    ros::NodeHandle nodeHandle;

    ros::Subscriber subscriber = nodeHandle.subscribe<std_msgs::String>("kaihei_room", 3, yao_callback);
    publisher = nodeHandle.advertise<std_msgs::String>("kaihei_room", 3);

    
    while(ros::ok)
    {
        ros::spinOnce();
    }
    return 0;
}

File: first_publisher_subscriber/scripts/yao_node.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String


def yao_callback(msg):
    global publisher
    if msg.data == "Play a game?":
        response_msg = String(data="+++++++++")
        publisher.publish(response_msg)
        
if __name__ == "__main__":
    rospy.init_node("yao_node")
    
    subscriber = rospy.Subscriber("kaihei_room", String, yao_callback)
    publisher = rospy.Publisher("kaihei_room", String, queue_size=3)
    
    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        rospy.spin()
        
        rate.sleep()